feature/sort/src/test/kotlin/com/merxury/blocker/feature/sort/ComponentSortViewModelTest.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2024 Blocker
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.merxury.blocker.feature.sort
+
+import com.merxury.blocker.core.model.preference.ComponentShowPriority
+import com.merxury.blocker.core.model.preference.ComponentSorting
+import com.merxury.blocker.core.model.preference.SortingOrder
+import com.merxury.blocker.core.testing.repository.TestUserDataRepository
+import com.merxury.blocker.core.testing.repository.defaultUserData
+import com.merxury.blocker.core.testing.util.MainDispatcherRule
+import com.merxury.blocker.feature.sort.ComponentSortInfoUiState.Success
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import kotlin.test.assertEquals
+
+class ComponentSortViewModelTest {
+    @get:Rule
+    val mainDispatcherRule = MainDispatcherRule()
+
+    private val userDataRepository = TestUserDataRepository()
+    private lateinit var viewModel: ComponentSortViewModel
+
+    @Before
+    fun setup() {
+        viewModel = ComponentSortViewModel(userDataRepository)
+    }
+
+    @Test
+    fun componentSortInfoUiState_whenInitial_thenShowDefault() = runTest {
+        assertEquals(ComponentSortInfoUiState.Loading, viewModel.componentSortInfoUiState.value)
+    }
+
+    @Test
+    fun componentSortInfoUiState_whenLoaded_thenShowSuccess() = runTest {
+        val collectJob = launch(UnconfinedTestDispatcher()) {
+            viewModel.componentSortInfoUiState.collect()
+        }
+
+        userDataRepository.sendUserData(defaultUserData)
+        assertEquals(
+            Success(defaultUserData.toComponentSortInfo()),
+            viewModel.componentSortInfoUiState.value,
+        )
+
+        collectJob.cancel()
+    }
+
+    @Test
+    fun componentSortInfoUiState_updateComponentSorting() = runTest {
```

Comment:
Follow given_when_then format

Reply:
componentSortInfoUiState_whenUpdateComponentSorting_thenUpdateComponentSorting ?
I think it's not necessary.